x/axelarnet/message_handler.go

```diff
@@ -129,17 +190,38 @@ func OnRecvMessage(ctx sdk.Context, k keeper.Keeper, ibcK keeper.IBCKeeper, n ty
 		Address: data.GetSender(),
 	}
 
+	// generate message id in advance to link fee payment and message
+	msgID := n.GenerateMessageID(ctx)
```

Comment:
I don't like burning a counter if send token was received. Maybe move this logic into the `pay` method? And it can be called from the two handlers without too much duplication.